ntoskrnl/io/pnpmgr/pnpmgr.c

```diff
@@ -1760,6 +1765,80 @@ IopGetParentIdPrefix(PDEVICE_NODE DeviceNode,
     return Status;
 }
 
+static
+BOOLEAN
+IopValidateID(
+    _In_ PWCHAR Id,
+    _In_ SIZE_T MaxIdLen,
+    _In_ ULONG MaxSeparators,
+    _In_ BOOLEAN IsMultiSz)
+{
+    PWCHAR PtrSymbol;
+    PWCHAR StringEnd;
+    WCHAR Symbol;
+    ULONG SeparatorsCount = 0;
+    PWCHAR PtrPrevSymbol = NULL;
+
+    PAGED_CODE();
+
+    StringEnd = Id + MaxIdLen;
+
+    for (PtrSymbol = Id; PtrSymbol < StringEnd; PtrSymbol++)
+    {
+        Symbol = *PtrSymbol;
+
+        if (Symbol == UNICODE_NULL)
+        {
+            if (!IsMultiSz || (PtrPrevSymbol && PtrSymbol == PtrPrevSymbol + 1))
+            {
+                if (MaxSeparators == SeparatorsCount ||
+                    MaxSeparators == MAX_SEPARATORS_MULTISZ)
+                {
+                    return TRUE;
+                }
+
+                DPRINT("IopValidateID: SeparatorsCount - %lu, MaxSeparators - %lu\n",
```

Comment:
All these failure cases can be DPRINT1's

Reply:
Changed